### PR TITLE
Add lan channel fact and user channel support

### PR DESCRIPTION
**Pull request by nathanlcarlson**
Based on https://github.com/jhoblitt/puppet-ipmi/pull/32 .

Intentionally named ::user parameter `channel` instead of `lan_channel` to be more general.

### CODE REVIEW

**Comment by jhoblitt** [comment]
@nathanlcarlson How would you feel about adding the channel as a structured fact. Something like `ipmi.channel`?

**Comment by jhoblitt** [comment]
> @jhoblitt Sure. Just this fact or should I create a structure for all `impi*` facts? The current flat facts could remain and I could add docs about them being deprecated?

I was only hoping to stop the growth of top level facts but if your willing to do the full conversion, that would be fantastic!

**Comment by nathanlcarlson** [comment]
Kk, I made some standalone gathering and structuring of the facts. I'm not sure how REFERENCE.md gets updated or if there are other PDK items to take care of.

**Comment by nathanlcarlson** [comment]
Ahh, I found the reference update command in the test results.

**Comment by nathanlcarlson** [comment]
It looks like these new failures are related to CentOS package mirrors? Maybe partly related to the retirement of CentOS 7?

**Comment by nathanlcarlson** [comment]
Those may be difficult to resolve. Any chance we can just ignore them?

**Comment by jhoblitt** [comment]
> Those may be difficult to resolve. Any chance we can just ignore them?

I think we should remove centos completely from `metadata.json` in another PR labeled as a `breaking-change`.  Do you want to take that on too? :pray:

**Comment by nathanlcarlson** [comment]
I did my best to track down this issue and opened this https://github.com/voxpupuli/beaker-docker/issues/144 and am trying the advice there.

**Comment by jhoblitt** [comment]
It's the plumbing in those module that needs to be updated.  I'll run msync on it shortly.

**Comment by jhoblitt** [comment]
FYI:
- #79
- #80 
- #81 
- #82

**Comment by jhoblitt** [comment]
@nathanlcarlson CI is passing on `master` now.  Could you try a rebase?

File: lib/facter/ipmi.rb [hunk 1]
```diff
@@ -79,3 +80,37 @@ def add_ipmi_fact(name, value)
   @channel_nr = channel
   IPMIChannel.new(@channel_nr).load_facts
 end
+
+Facter.add(:ipmi) do
+  confine kernel: 'Linux'
+  setcode do
+    ipmi_hash = {}
```
Comment (jhoblitt) [inline lib/facter/ipmi.rb:87]: This all looks good except I have a quibble with using Hungarian notation.

**Review by jhoblitt — CHANGES_REQUESTED** [review]

**Review by jhoblitt — APPROVED** [review]
LGTM. Thank you for doing the fact conversion!

**Comment by jhoblitt** [comment]
Released as https://github.com/jhoblitt/puppet-ipmi/tree/v6.0.0

**Pull request merged**